### PR TITLE
FOUNDATIONS: Remove Team

### LINKED ISSUE #132: FOUNDATIONS: Remove `Team`

### PR DESCRIPTION
**Pull request by jasurbekyusuf**
Closes #132

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Exception.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            // then
+            actualTeamDependencyValidationException.Should().BeEquivalentTo(
+                expectedTeamDependencyValidationException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectTeamByIdAsync(It.IsAny<Guid>()), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+            expectedTeamDependencyValidationException))), Times.Once);
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Exception.RemoveById.cs:55]: Tab

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Exception.RemoveById.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            var failedTeamStorageException =
+                new FailedTeamStorageException(sqlException);
+
+            var expectedTeamDependencyException =
+                new TeamDependencyException(failedTeamStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectTeamByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+            // when
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Exception.RemoveById.cs:81]: need empty line

**Review by jamshidbek001 — CHANGES_REQUESTED** [review]
correct the mistakes

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Logic.RemoveById.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+            // when
+            Team actualTeam = await this.teamService
+                .RemoveTeamByIdAsync(inputTeamId);
+
+            // then
+            actualTeam.Should().BeEquivalentTo(expectedTeam);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectTeamByIdAsync(inputTeamId), Times.Once());
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Teams/TeamServiceTests.Logic.RemoveById.cs:46]: Once() => Once ga o'zgartirasiz

**Review by Javlonbek-dev — APPROVED** [review]

File: Tarteeb.Api/Services/Foundations/Teams/ITeamService.cs [hunk 4]
```diff
@@ -13,5 +13,6 @@ public interface ITeamService
     {
         ValueTask<Team> AddTeamAsync(Team team);
         ValueTask<Team> RetrieveTeamByIdAsync(Guid teamId);
+        ValueTask<Team> RemoveTeamByIdAsync(Guid TeamId);
```
Comment (StudentofTATU) [inline Tarteeb.Api/Services/Foundations/Teams/ITeamService.cs:16]: change variable name teamId

File: Tarteeb.Api/Services/Foundations/Teams/TeamService.cs [hunk 5]
```diff
@@ -48,5 +48,18 @@ public partial class TeamService : ITeamService
 
             return maybeTeam;
         });
+
+        public ValueTask<Team> RemoveTeamByIdAsync(Guid teamId) =>
+        TryCatch(async () =>
+        {
+            ValidateTeamId(teamId);
+
+            Team maybeTeam = await this.storageBroker
```
Comment (StudentofTATU) [inline Tarteeb.Api/Services/Foundations/Teams/TeamService.cs:57]: await this.storageBroker     buni pastdagi qatorga tushiring.

**Review by StudentofTATU — CHANGES_REQUESTED** [review]
make some changes

**Review by StudentofTATU — APPROVED** [review]

**Review by iamDiyorjon — COMMENTED** [review]
LGTM

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 6]
```diff
@@ -43,7 +43,7 @@ public async ValueTask<ActionResult<User>> PostUserAsync(User user)
             catch (UserDependencyValidationException userDependencyValidationException)
             {
                 return BadRequest(userDependencyValidationException.InnerException);
-                
+
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Controllers/UsersController.cs:46]: Remove this line

File: Tarteeb.Api/Services/Foundations/Teams/TeamService.cs [hunk 7]
```diff
@@ -48,5 +48,18 @@ public partial class TeamService : ITeamService
...
+
+        public ValueTask<Team> RemoveTeamByIdAsync(Guid teamId) =>
+        TryCatch(async () =>
+        {
+            ValidateTeamId(teamId);
+
+            Team maybeTeam =
+                await this.storageBroker.SelectTeamByIdAsync(teamId);
+
+            ValidateStorageTeam(maybeTeam, teamId);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Foundations/Teams/TeamService.cs:60]: Mana shuni `ValidateStorageTeamExists` ga o'zgartiring. Chunki bu metodni qilayotgan ishi shu bo'lyapti

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**